### ISSUE TITLE
Is there a way to set the uncertainty of the observations for conditioning ?

### ISSUE THREAD

**Issue by Galto2000** [cornellius-gp/gpytorch#890]
Howdy folks,

Say we trained a GP on data with some known standard deviation `d1`. Then we deploy this GP in the world, but with sensors that might be noisier that the one from which the training data was obtained: `d2 > d1`. 

I am using  `model.set_train_data(obs_x, obs_y)` for performing the conditioning on the (noisier) observations, and then I get my predictive distribution as follows: 

```
    with torch.no_grad(), gpytorch.settings.fast_pred_var():
        test_x = torch.linspace(1000, 0, 100)
        predictions = likelihood(model(test_x))
```

Where do I need to account for the noise of the observations (d2)? Do I set the likelihood to d2?

Thanks in advance

Galto

**Comment by jacobrgardner** [comment]
Yeah, the likelihood noise would correspond to the known observation noise.

You could also just do `pred = model(test_x)` and then `real_variance = pred.variance + d2 ** 2`, since that's all tthe likelihood is really doing in the regression setting.

**Comment by Galto2000** [comment]
Thanks @jacobrgardner 

How would I go about two, or more, sets of observations, each with different noise: say we have different sensors measuring the same property, but at different accuracies. Then I would like to condition my model on all observation sets in order to get a predictive posterior distribution, in a sensor-fusion kinda of way?

**Comment by jacobrgardner** [comment]
You probably want to just write a new likelihood for that. If your data is sorted reasonably so like `[All data from source one; then all data from source 2]`, you basically want something like the fixed noise likelihood that adds `DiagLazyTensor([d1, d1, ..., d1, d2, d2, ..., d2])` to the covariance matrix.

**Comment by Galto2000** [comment]
I have pretty good control over the sensors, so it can be done that way.

Thanks for the advice Jacob!

**Comment by Galto2000** [comment]
How would I go about doing the same, but for a multi-task GP (say two outputs)?

**Comment by Galto2000** [comment]
Hi  @jacobrgardner

I think I am missing something or misunderstood your explanation.  The way I am interpreting your explanation is that the measurement noise is added after the prediction (i.e. conditioning) has been done. But I always understood that conditioning (with noise) is done by replacing `Kxx` with `Kxx + sigma^2*I` in the equations for the predictive mean and predictive covariance. So somehow the measurement noise have to get "inserted" before doing the prediction, no?

![GP Equations Rasmussen](https://user-images.githubusercontent.com/19415965/66368764-8db71500-e967-11e9-950b-3f02c86d426d.PNG)

Your explanation about organizing the data made perfect sense to me, and so if I wanted to do multi-sensor fusion/prediction , for n sensors each with a sigma (`sigma_1^2, sigma_2^2 .. sigma_n^2`) then I would need to do something like this:

![GP Sensor Fustion 2](https://user-images.githubusercontent.com/19415965/66368867-f8685080-e967-11e9-9c21-2e5de5a99ebb.PNG)

I am not understanding how adding `DiagLazyTensor([d1, d1, ..., d1, d2, d2, ..., d2]` to the prediction is the equivalent.

On top of all that, I want to do this over multi-task GP regression, since each sensor is measuring various properties ( all at the same locations, but with different sensors and thus different noises), like temperature,pressure, % water vapour, etc.
